src/main/java/org/generationcp/middleware/api/germplasm/GermplasmServiceImpl.java

```diff
@@ -417,14 +419,26 @@ public long countAttributesByGUID(final String gemrplasmUUID, final List<String>
 		return this.daoFactory.getAttributeDAO().countAttributesByGUID(gemrplasmUUID, attributeDbIds);
 	}
 
-  private List<Integer> getGidsOfGermplasmWithDerivativeOrMaintenanceDescendants(final List<Germplasm> germplasmList) {
-	  final List<Integer> gids = germplasmList.stream().map(Germplasm::getGid).collect(Collectors.toList());
+	@Override
+	public Germplasm getGermplasmWithPreferredName(final Integer gid) {
```

Comment:
It is not ideal to return the raw `Germplasm` Hibernate entity. 
Check if you can use/adjust  `GermplasmDto getGermplasmDtoById(final Integer gid)`.
I noted we need the following info in `GermplasmTreeNode` constructor:
- preferred name (already being retrieved)
- method name (already bring retrieved)
- method code

Reply:
If we can indeed reuse, no need for this `getGermplasmWithPreferredName`

Reply:
There's no need for getGermplasmWithPreferredName and getProgenitorsWithPreferredName. I already removed them.   I used GermplasmDAO.getById() to get the germplasm in GermplasmPedigreeServiceImpl. The method name, method code and preferred name are already available there.